numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -4255,12 +4255,23 @@ initialize_numeric_types(void)
     PyStringArrType_Type.tp_alloc = NULL;
     PyStringArrType_Type.tp_free = NULL;
 
+    /*
+     * NOTE: these slots aren't actually being used for
+     * string and unicode NumPy arrays -- the repr handling
+     * appears to funnel through np.ndarray Type struct and indeed
+     * none of the functions pointed to here even exist; trying
+     * to create them has no effect on behavior -- they
+     * are currently circumvented
+     */
```

Comment:
The functions exist, they come from `scalartypes.c.src`, line 474 and are generated.  When I run tests under a debugger, I hit `unicodetype_str` quite often in `test_arrayprint.py`